File: C#/Sln/vs2013/NET005 Combobox下拉组合框/WinMain.cs
using System;
using System.Collections.Generic;
//using System.Linq;
using System.Text;
//using System.Threading.Tasks;
using System.Runtime.InteropServices;
using XCGUI;


class CWindow
{
    unsafe public void Create()
    {
        XC.XInitXCGUI("");
        int hWindow = XC.XWnd_Create(0, 0, 500, 300, "xcgui", 0, 15);

        int hList = XC.XList_Create(0, 0, 300, 200, hWindow);
        XC.XList_SetItemTemplateXML(hList, "List_Item.xml");
        XC.XList_AddColumn(hList, 100);
        XC.XList_AddColumn(hList, 100);

        int hHead = XC.XAdapterMap_Create();
        XC.XList_BindAdapterHeader(hList, hHead);
        XC.XAdapterMap_AddItemText(hHead, "name", "第一列");
        XC.XAdapterMap_AddItemText(hHead, "name2", "第二列");


        int body = XC.XAdapterTable_Create();
        XC.XList_BindAdapter(hList, body);
        XC.XAdapterTable_AddColumn(body, "name");
        XC.XAdapterTable_AddColumn(body, "name2");

        int nItem = XC.XAdapterTable_AddItemText(body, "1.1");
        XC.XAdapterTable_SetItemText(body, nItem, 1, "1.2");


        XC.XWnd_ShowWindow(hWindow, 5);
        XC.XRunXCGUI();
        XC.XExitXCGUI();
    }
}

namespace Project1
{
    class RunMain
    {
        static void Main()
        {
            CWindow pWindow = new CWindow();
            pWindow.Create();
        }

    }
}

File: VS/Sln/vc6/VC6_093 List-列表元素 动态修改项目文本/WinMain.cpp
#include "Common.h"

class CMyWindowList
{
public:
    HWINDOW m_hWindow;
    HELE    m_hList;
	HXCGUI hAdapter;
    CMyWindowList()
    {
        Init();
    }
	int OnBtnClick(BOOL *pbHandled)
	{
		//此种方法，修改数据适配器保存的数据
		XAdapterTable_SetItemText(hAdapter,0,0,L"修改1");
		XAdapterTable_SetItemText(hAdapter,0,1,L"修改2");
		XAdapterTable_SetItemText(hAdapter,0,2,L"修改3");
		XList_RefreshData(m_hList);
		XEle_RedrawEle(m_hList);

		return 0;
	}
    void Init()
    {
        m_hWindow = XWnd_Create(0, 0, 550, 380, L"炫彩界面库窗口",NULL, xc_window_style_default);
        XBtn_SetType(XBtn_Create(5, 3, 60, 20, L"Close", m_hWindow),button_type_close);
		
		HELE hBtn = XBtn_Create(100,3,200,20,L"动态修改项目文本",m_hWindow);
		XEle_RegEventCPP(hBtn,XE_BNCLICK,&CMyWindowList::OnBtnClick);

        m_hList= XList_Create(20,40,500,300,m_hWindow);
        XList_SetItemTemplateXML(m_hList,L"List_Item.xml");
        XSView_SetLineSize(m_hList,20,20);
		
        XList_AddColumn(m_hList,100);
        XList_AddColumn(m_hList,100);
        XList_AddColumn(m_hList,100);
		
        HXCGUI hAdapterHeader=XAdapterMap_Create();
        XList_BindAdapterHeader(m_hList,hAdapterHeader);
        XAdapterMap_AddItemText(hAdapterHeader,(XC_NAME),L"aaa");
        XAdapterMap_AddItemText(hAdapterHeader,(XC_NAME2),L"bbb");
        XAdapterMap_AddItemText(hAdapterHeader,(XC_NAME3),L"ccc");
        XAdapterMap_AddItemText(hAdapterHeader,(XC_NAME4),L"test");
		
        hAdapter=XAdapterTable_Create();
        XList_BindAdapter(m_hList,hAdapter);
        XAdapterTable_AddColumn(hAdapter,(XC_NAME));
        XAdapterTable_AddColumn(hAdapter,(XC_NAME2));
        XAdapterTable_AddColumn(hAdapter,(XC_NAME3));
		
        XList_SetColumnWidth(m_hList,0,150);
        XList_SetColumnWidth(m_hList,1,150);
        XList_SetColumnWidth(m_hList,2,150);
		
        wchar_t tmp[MAX_PATH]={0};
        for (int i=0;i<20;i++)
        {
            wsprintf(tmp,L"item-%d",i);
            XAdapterTable_AddItemText(hAdapter,tmp);
			
            wsprintf(tmp,L"child-%d-1",i);  XAdapterTable_SetItemText(hAdapter,i,1,tmp);
            wsprintf(tmp,L"child-%d-2",i);  XAdapterTable_SetItemText(hAdapter,i,2,tmp);
        }
		
        XWnd_ShowWindow(m_hWindow,SW_SHOW);
    }
};

int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPTSTR lpCmdLine, int nCmdShow)
{
    XInitXCGUI();
    CMyWindowList  MyWindow;
    XRunXCGUI();
    XExitXCGUI();
    return 0;
}
